tests/functional/ipset_controller_test.go

```diff
@@ -312,6 +314,72 @@ var _ = Describe("IPSet controller", func() {
 		})
 	})
 
+	When("an IPSet with multiple networks gets created", func() {
+		var netSpecs []networkv1.Network
+		var networks []networkv1.IPSetNetwork
+		BeforeEach(func() {
+			netSpecs = []networkv1.Network{}
+			netSpecs = append(netSpecs, GetNetSpec(net1, GetSubnet1(subnet1)))
+			netSpecs = append(netSpecs, GetNetSpec(net2, GetSubnet2(subnet1)))
+			netSpecs = append(netSpecs, GetNetSpec(net3, GetSubnet3(subnet1)))
+			networks = []networkv1.IPSetNetwork{}
+			networks = append(networks, GetIPSetNet1Lower())
+			networks = append(networks, GetIPSetNet2())
+			networks = append(networks, GetIPSetNet3())
+
+			Expect(networks).To(HaveLen(len(netSpecs)))
+			// we want to ensure order is preserved so we will randomize the order of the networks
+			// and then create the IPSet
+			rand.Shuffle(len(networks), func(i, j int) {
+				networks[i], networks[j] = networks[j], networks[i]
+				netSpecs[i], netSpecs[j] = netSpecs[j], netSpecs[i]
+			})
+			// confirm that after we randomize the order the networks and netSpec are still in sync
+			for i := 0; i < len(networks); i++ {
+				Expect(networks[i].Name).To(Equal(netSpecs[i].Name))
+			}
+			netCfg := CreateNetConfig(namespace, GetNetConfigSpec(netSpecs...))
+			ipset := CreateIPSet(namespace, GetIPSetSpec(false, networks...))
+			DeferCleanup(func(_ SpecContext) {
+				th.DeleteInstance(ipset)
+				th.DeleteInstance(netCfg)
+			}, NodeTimeout(timeout))
+
+			ipSetName = types.NamespacedName{
+				Name:      ipset.GetName(),
+				Namespace: namespace,
+			}
+			netCfgName.Name = netCfg.GetName()
+			netCfgName.Namespace = netCfg.GetNamespace()
+			Eventually(func(g Gomega) {
+				res := GetNetConfig(netCfgName)
+				g.Expect(res).ToNot(BeNil())
+			}, timeout, interval).Should(Succeed())
+
+			th.ExpectCondition(
+				ipSetName,
+				ConditionGetterFunc(IPSetConditionGetter),
+				condition.ReadyCondition,
+				corev1.ConditionTrue,
+			)
+		})
+
+		It("it should have created an IPSet with multiple networks", func() {
+
+			instance := &networkv1.IPSet{}
+
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, ipSetName, instance)).Should(Succeed())
+				for i := 0; i < len(networks); i++ {
+					g.Expect(instance.Spec.Networks[i].Name).To(Equal(networks[i].Name))
+					g.Expect(instance.Spec.Networks[i].Name).To(Equal(netSpecs[i].Name))
```

Comment:
Don't think this is valid/needed. it works for this test implementation, but could be different in a real env. This validates that the network referenced in instance.Spec.Networks[i] is at the same place as in the osnetconfig. Lets assume you create the netconfig with 3 networks, but the ipset just requests 2 of them, so it could be index 1 in Networks, but index 2 in the osnetconfig.